assert.js

```diff
@@ -596,6 +596,51 @@ assert.ifError = function ifError(err) {
   }
 };
 
+// Currently in sync with Node.js lib/assert.js
+// https://github.com/nodejs/node/commit/2a871df3dfb8ea663ef5e1f8f62701ec51384ecb
+function internalMatch(string, regexp, message, fn) {
+  if (!isRegExp(regexp)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'regexp', 'RegExp', regexp
+    );
+  }
+  const match = fn.name === 'match';
```

Comment:
function names are configurable, and some browsers don't support function names at all (IE 9-11). Instead, this should probably just be a hardcoded string.

Reply:
True, updated accordingly. However, I'm wondering if we should update the rest of the `fn.name`s in the file too, like in https://github.com/browserify/commonjs-assert/blob/bba838e9ba9e28edf3127ce6974624208502f6bc/assert.js#L337 in a follow-up PR?

Reply:
Yes, we definitely should. Relying on function names is not robust.